src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
+import './TodoList.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import classname from 'classnames/bind';
+import { deleteTodo, getTodos, getUserById } from '../../api/api';
+import { setDeleteTodoAction, setTodosAction, setUserAction } from '../../store/actions';
+import { getTodosSeclectors, OptionValue } from '../../store/selectors';
+
+export const TodoList:React.FC = () => {
+  const dispatch = useDispatch();
+  const [query, setQuery] = useState('');
+  const [optionValue, setOptionValue] = useState(' ');
+  const todos = useSelector(getTodosSeclectors(optionValue, query));
+
+  useEffect(() => {
+    const loadTodosFromServer = async () => {
+      const todosFromServer = await getTodos();
+
+      dispatch(setTodosAction(todosFromServer));
+    };
+
+    loadTodosFromServer();
+  }, []);
+
+  const getUser = async (id: number) => {
+    const userFromServer = await getUserById(id);
+
+    dispatch(setUserAction(userFromServer));
+  };
+
+  const deleteTodoFromServer = async (id: number) => {
+    await deleteTodo(id);
+    const todosFromServer = await getTodos();
+
+    dispatch(setDeleteTodoAction(todosFromServer));
```

Comment:
will be better to call here add todos action, cause you have the same functionality for both cases